radialgraph/src/main/java/com/duartbreedt/radialgraph/drawable/GraphDrawable.kt

```diff
@@ -29,11 +34,67 @@ abstract class GraphDrawable(
 
         return Paint().apply {
             strokeWidth = graphConfig.strokeWidth
-            color = state.color
             pathEffect = DashPathEffect(floatArrayOf(state.length!!, state.length!!), phase)
             style = Paint.Style.STROKE
             flags = Paint.ANTI_ALIAS_FLAG
             strokeCap = graphConfig.capStyle.paintCapStyle
+
+            if (!graphConfig.isGradientEnabled()) {
+                color = state.color
+            }
+
+            if (graphConfig.gradientType == GradientType.SWEEP) {
+                applySweepGradient(state, this, graphConfig.gradientFill)
+            }
+        }
+    }
+
+    private fun applySweepGradient(state: SectionState, paint: Paint, gradientFill: GradientFill) {
+        if (state.gradientColors.isNullOrEmpty()) {
+            paint.color = state.color
+            return
+        }
+
+        val colorList = mutableListOf(
+            state.color
+        )
+        colorList.addAll(state.gradientColors)
+
+        var positionList: MutableList<Float>? = null
+        if (gradientFill == GradientFill.SECTION) {
+            positionList = MutableList(colorList.size) { it.toFloat() }
+            //Get spacing between each gradient section
+            val gradientGap = (1f / (positionList.size -1)) * state.sweepSize
+
+            positionList = positionList.map {
+                (gradientGap * it) + state.startPosition
+            }.toMutableList()
+
+            //Fix for gradient overflow when using a cap style other than 'BUTT'
+            if (state.startPosition == 0f && graphConfig.capStyle != Cap.BUTT) {
+                colorList.add(state.color)
+                positionList.addAll(
+                    listOf(
+                        positionList.removeLast().coerceAtMost(endGradientCapFill),
+                        endGradientCapFill
+                    )
+                )
+            }
+        }
+
+        paint.shader = SweepGradient(
+            this@GraphDrawable.bounds.centerX().toFloat(),
+            this@GraphDrawable.bounds.centerY().toFloat(),
```

Comment:
Dangerous use of bounds. We should ideally use `calculateBoundaries` since it accounts for the stroke width.